src/vstest.console/CommandLine/Interfaces/IInferHelper.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.CommandLine.Processors
+{
+    using System.Collections.Generic;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+    /// <summary>
+    /// Defines interface for infering run configuration values.
+    /// </summary>
+    internal interface IInferHelper
```

Comment:
I think this interface doesn't add any value. There will always be one implementation, and that seems to be a singleton.